app/controllers/users_controller.rb

```diff
@@ -0,0 +1,69 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: %i[ show edit update destroy ]
```

Comment:
- Before action là gì?
- %i có ý nghĩa gì?

Reply:
- Tổng quan về ActiveSupport: Đây là một module trong Rails cung cấp các công cụ cần thiết cho phép các module khác nhau sau khi include ActiveSupport có thể tạo callback riêng cho nó.
- Trong Active Support khai báo 3 phương thức quan trọng và cốt lõi để có thể tạo nên một callback
+ define_callback: Định nghĩa ra các sự kiện trong một chu kì hoạt động của đối tượng sẽ được hỗ trợ callback, vdu như save, destroy, update, ... là các sự kiện khá phổ biến, với define_callback ta có thể tự custom một callback cho riêng mình.
+ set_callback: Thiết lập các instance method hoặc proc, ... để sẵn sàng được gọi lại, có thể hiểu là install các phương thức đã khai báo trước đó và sẵn sàng đợi cho đến khi được callback.
+ run_callback: Chạy các phương thức đã được install trước đó bởi set_callback vào một thời điểm thích hợp.
- Có 3 loại callback được hỗ trợ đó là before, after và around. before callback chạy trước một sự kiện, after callback chạy sau khi sự kiện xảy ra và around callback chạy cả trước và sau khi sự kiện xảy ra.


%w : Array of Strings
%i : Array of Symbols
%q : String
%r : Regular Expression
%s : Symbol
%x : Backtick (capture subshell result)